### PR TITLE
[14.0][FW] [13.0] Change sorting of locations in the pack putaway computation

### PR DESCRIPTION
**Pull request by sebalix**
Port of #237 and #240 from 13.0 to 14.0.

### CODE REVIEW

**Review by TDu — APPROVED** [review]

**Review by jgrandguillaume — APPROVED** [review]
LGTM

**Review by jbaudoux — APPROVED** [review]

File: stock_storage_type/models/stock_location.py [hunk 1]
```diff
@@ -379,10 +383,55 @@ def get_storage_locations(self, products=None):
         locations = self.browse()
         if self.pack_putaway_strategy == "none":
             locations = self
-        elif self.pack_putaway_strategy == "ordered_locations":
-            locations = self._get_ordered_leaf_locations()
+        else:
+            locations = self._get_sorted_leaf_locations(products)
```
Comment (jbaudoux) [inline stock_storage_type/models/stock_location.py:388]: There is a fix for this here https://github.com/OCA/wms/pull/280/files

Comment (sebalix) [reply stock_storage_type/models/stock_location.py:388]: Fix cherry-picked

**Pull request merged**